respy/shared.py

```diff
@@ -23,31 +24,52 @@ def _aggregate_keane_wolpin_utility(
     return value_function, flow_utility
 
 
-def get_conditional_probabilities(type_shares, initial_level_of_education):
-    """Calculate the conditional choice probabilities.
+@guvectorize(
+    ["f8[:, :], f8[:], f8[:]"],
+    "(n_classes, n_covariates), (n_covariates) -> (n_classes)",
+    nopython=True,
+    target="parallel",
+)
+def predict_multinomial_logit(coefficients, covariates, probs):
+    """Predict probabilities based on a multinomial logit regression.
 
-    The calculation is based on the multinomial logit model for one particular initial
-    condition.
+    The function is used to predict the probability for all types based on the initial
+    characteristics of an individual. This is necessary to sample initial types in the
+    simulation and to weight the probability of an observation by types in the
+    estimation.
+
+    The `multinomial logit model
+    <https://en.wikipedia.org/wiki/Multinomial_logistic_regression>`_  predicts the
+    probability that an individual belongs to a certain type. The sum over all
+    type-probabilities is one.
 
     Parameters
     ----------
-    type_shares : numpy.ndarray
-        Undocumented parameter.
-    initial_level_of_education : numpy.ndarray
-        Array with shape (n_obs,) containing initial levels of education.
+    coefficients : numpy.ndarray
+        Array with shape (n_classes, n_covariates).
+    covariates : numpy.ndarray
+        Array with shape (n_covariates,).
+
+    Returns
+    -------
+    probs : numpy.ndarray
+        Array with shape (n_classes,) containing the probabilities for each type.
 
     """
-    type_shares = type_shares.reshape(-1, 2)
-    covariates = np.column_stack(
-        (np.ones(initial_level_of_education.shape[0]), initial_level_of_education > 9)
-    )
-    probs = np.exp(covariates.dot(type_shares.T))
-    probs /= probs.sum(axis=1, keepdims=True)
+    n_classes, n_covariates = coefficients.shape
+
+    denominator = 0
 
-    if initial_level_of_education.shape[0] == 1:
-        probs = probs.ravel()
+    for type_ in range(n_classes):
```

Comment:
sometimes we use typ, sometimes type_ when we loop over types. Maybe we should align this in the long run. I'm indifferent which one we take.

Reply:
I am in favor of ``type_``. I am going to change everything to this.